libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -22,900 +22,66 @@
-}
-
-void YamlConfiguration::raiseComponentError(const std::string &component_name, const std::string &yaml_section, const std::string &reason) const {
-  std::string err_msg = "Unable to parse configuration file for component named '";
-  err_msg.append(component_name);
-  err_msg.append("' because " + reason);
-  if (!yaml_section.empty()) {
-    err_msg.append(" [in '" + yaml_section + "' section of configuration file]");
-  }
-
-  logging::LOG_ERROR(logger_) << err_msg;
-
-  throw std::invalid_argument(err_msg);
-}
-
-std::string YamlConfiguration::getOrGenerateId(const YAML::Node& yamlNode, const std::string& idField) {
-  std::string id;
-  auto node = yamlNode.as<YAML::Node>();
-
-  if (node[idField]) {
-    if (YAML::NodeType::Scalar == node[idField].Type()) {
-      id = node[idField].as<std::string>();
-      addNewId(id);
-      return id;
-    }
-    throw std::invalid_argument("getOrGenerateId: idField is expected to reference YAML::Node of YAML::NodeType::Scalar.");
-  }
-
-  id = id_generator_->generate().to_string();
-  logger_->log_debug("Generating random ID: id => [%s]", id);
-  return id;
-}
-
-std::string YamlConfiguration::getRequiredIdField(const YAML::Node& yaml_node, std::string_view yaml_section, std::string_view error_message) {
-  yaml::checkRequiredField(yaml_node, "id", yaml_section, error_message);
-  auto id = yaml_node["id"].as<std::string>();
-  addNewId(id);
-  return id;
-}
-
-YAML::Node YamlConfiguration::getOptionalField(const YAML::Node& yamlNode, const std::string& fieldName, const YAML::Node& defaultValue, const std::string& yamlSection,
-                                               const std::string& providedInfoMessage) {
-  std::string infoMessage = providedInfoMessage;
-  auto result = yamlNode.as<YAML::Node>()[fieldName];
-  if (!result) {
-    if (infoMessage.empty()) {
-      // Build a helpful info message for the user to inform them that a default is being used
-      infoMessage =
-          yamlNode.as<YAML::Node>()["name"] ?
-              "Using default value for optional field '" + fieldName + "' in component named '" + yamlNode.as<YAML::Node>()["name"].as<std::string>() + "'" :
-              "Using default value for optional field '" + fieldName + "' ";
-      if (!yamlSection.empty()) {
-        infoMessage += " [in '" + yamlSection + "' section of configuration file]: ";
-      }
-
-      infoMessage += defaultValue.as<std::string>();
-    }
-    logging::LOG_INFO(logger_) << infoMessage;
-    result = defaultValue;
+std::unique_ptr<core::ProcessGroup> YamlConfiguration::getYamlRoot(std::istream &yamlConfigStream) {
+  try {
+    YAML::Node rootYamlNode = YAML::Load(yamlConfigStream);
+    flow::Node root{std::make_shared<YamlNode>(rootYamlNode)};
+    return getRootFrom(root);
+  } catch (const YAML::ParserException &pe) {
+    logger_->log_error(pe.what());
+    std::rethrow_exception(std::current_exception());
   }
-
-  return result;
+  return nullptr;
```

Comment:
This is unreachable code (also at line 84).  Does one of our supported compilers insist on it?

Reply:
I don't know if any compiler complains, removed it